test/kpca.jl

```diff
@@ -0,0 +1,88 @@
+using MultivariateStats
+using Base.Test
+
+srand(34568)
+
+## data
+n = 10
+d = 5
+X = randn(d, n)
+
+# kernel calculations
+K = MultivariateStats.pairwise((x,y)->norm(x-y), X, X[:,1:2])
+@test size(K) == (n, 2)
+@test K[1,1] == 0
+@test K[3,2] == norm(X[:,3] - X[:,2])
+
+K = MultivariateStats.pairwise((x,y)->norm(x-y), X[:,1:3], X)
+@test size(K) == (3, n)
+@test K[1,1] == 0
+@test K[3,2] == norm(X[:,2] - X[:,3])
+
+K = similar(X, n, n)
+MultivariateStats.pairwise!(K, (x,y)->norm(x-y), X)
+@test size(K) == (n, n)
+@test K[1,1] == 0
+@test K[2,1] == norm(X[:,2] - X[:,1])
+
+KC = fit(MultivariateStats.KernelCenter, K)
+Iₙ = ones(n,n)/n
+@test MultivariateStats.transform!(KC, copy(K)) ≈ K - Iₙ*K - K*Iₙ + Iₙ*K*Iₙ
+
+K = MultivariateStats.pairwise((x,y)->norm(x-y), X, X[:,1])
+@test size(K) == (n, 1)
+@test K[1,1] == 0
+@test K[2,1] == norm(X[:,2] - X[:,1])
+
+KC = fit(MultivariateStats.KernelCenter, K)
+@test all(x->x==KC.total, MultivariateStats.transform!(KC, copy(K)))
+
+
+## check different parameters
+X = randn(d, n)
+M = fit(KernelPCA, X, maxoutdim=d)
+M2 = fit(PCA, X, method=:cov, pratio=1.0)
+@test indim(M) == d
+@test outdim(M) == d
+@test abs.(transform(M, X)) ≈ abs.(transform(M2, X))
```

Comment:
Do you know about `Base.Test.test_approx_eq_modphase`? It can be useful to check the results taking into account possible sign changes.

Reply:
`test_approx_eq_modphase` checks if the norm of difference (sum) is close to zero. In the above case, different methods produce same transformations but orthogonally rotated, so the difference norm in such case could take a large value.